openhtf/plugs/user_input.py

```diff
@@ -173,26 +111,67 @@ def run(self):
 
 class UserInput(plugs.BasePlug):
   """Get user input from inside test phases."""
+  def __init__(self):
+    self._prompt = None
+    self._response = None
+    self._cond = threading.Condition()
+    
   def prompt(self, message, text_input=False, timeout_s=None):
-    return PROMPT_MANAGER.DisplayPrompt(
-        message, text_input=text_input, timeout_s=timeout_s)
+    """Prompt for a user response by showing the message.
+
+    Args:
+      message: The message to display to the user.
+      text_input: True iff the user needs to provide a string back.
+    Returns:
+      The string input by the user.
+    """
+    timeout_s = timeout_s or DEFAULT_TIMEOUT_S
+    with self._cond:
+      if self._prompt is not None:
+        self._prompt = None
+        raise MultiplePromptsError
+      self._prompt = Prompt(id=uuid.uuid4(),
+                           message=message,
+                           text_input=text_input)
+      self._response = None
+      _LOG.debug('Displaying prompt (%s): "%s"%s%s', self._prompt.id,
+                 message, ', Expects text' if text_input else '',
+                 ', Timeout: %s sec' % timeout_s if timeout_s else '')
+
+      console_prompt = ConsolePrompt(
+          message, functools.partial(self.respond, self._prompt.id))
+      console_prompt.start()
+      self._cond.wait(timeout_s)
+      console_prompt.Stop()
+      self._prompt = None
+      if self._response is None:
+        self._prompt = None
+        raise PromptUnansweredError
+      return self._response
+
+  def respond(self, prompt_id, response):
+    """Respond to the prompt that has the given ID.
+
+    If there is no active prompt or the prompt id being responded to doesn't
+    match the active prompt, do nothing.
+    """
+    _LOG.debug('Responding to prompt (%s): "%s"', prompt_id, response)
+    with self._cond:
+      if self._prompt is not None and prompt_id == self._prompt.id:
+        self._response = response
+        self._cond.notifyAll()
+        return True  # The response was used.
+      return False  # The response was not used.
 
 
 def prompt_for_test_start(
     message='Provide a DUT ID in order to start the test.', text_input=True,
     timeout_s=60*60*24):
   """Make a test start trigger based on prompting the user for input."""
   def trigger():  # pylint: disable=missing-docstring
-    return PROMPT_MANAGER.DisplayPrompt(
+    plug = UserInput()
+    response = plug.prompt(
         message, text_input=text_input, timeout_s=timeout_s)
+    plug.tearDown()
```

Comment:
we might consider adding some utility to plugs' usage outside of the plug manager, auto-teardown on exit from a context, for example